cns/restserver/internalapi_test.go

```diff
@@ -235,15 +234,42 @@ func TestReconcileNCWithExistingState(t *testing.T) {
 	}
 	req := generateNetworkContainerRequest(secondaryIPConfigs, "reconcileNc1", "-1")
 
-	expectedAllocatedPods := make(map[string]cns.KubernetesPodInfo)
-	expectedAllocatedPods["10.0.0.6"] = cns.KubernetesPodInfo{
-		PodName:      "reconcilePod1",
-		PodNamespace: "PodNS1",
+	expectedAllocatedPods := map[string]cns.PodInfo{
+		"10.0.0.6": cns.NewPodInfo("", "", "reconcilePod1", "PodNS1"),
+		"10.0.0.7": cns.NewPodInfo("", "", "reconcilePod2", "PodNS1"),
 	}
 
-	expectedAllocatedPods["10.0.0.7"] = cns.KubernetesPodInfo{
-		PodName:      "reconcilePod2",
-		PodNamespace: "PodNS1",
+	expectedNcCount := len(svc.state.ContainerStatus)
+	returnCode := svc.ReconcileNCState(&req, expectedAllocatedPods, fakes.NewFakeScalar(releasePercent, requestPercent, batchSize), fakes.NewFakeNodeNetworkConfigSpec(initPoolSize))
+	if returnCode != Success {
+		t.Errorf("Unexpected failure on reconcile with no state %d", returnCode)
+	}
+
+	validateNCStateAfterReconcile(t, &req, expectedNcCount+1, expectedAllocatedPods)
+}
+
+func TestReconcileNCWithExistingStateFromInterfaceID(t *testing.T) {
+	restartService()
+	setEnv(t)
+	setOrchestratorTypeInternal(cns.KubernetesCRD)
+	cns.GlobalPodInfoScheme = cns.InterfaceIDPodInfoScheme
+	defer func() { cns.GlobalPodInfoScheme = cns.KubernetesPodInfoScheme }()
+
+	secondaryIPConfigs := make(map[string]cns.SecondaryIPConfig)
+
+	var startingIndex = 6
+	for i := 0; i < 4; i++ {
+		ipaddress := "10.0.0." + strconv.Itoa(startingIndex)
+		secIpConfig := newSecondaryIPConfig(ipaddress, -1)
+		ipId := uuid.New()
+		secondaryIPConfigs[ipId.String()] = secIpConfig
+		startingIndex++
+	}
+	req := generateNetworkContainerRequest(secondaryIPConfigs, "reconcileNc1", "-1")
+
+	expectedAllocatedPods := map[string]cns.PodInfo{
+		"10.0.0.6": cns.NewPodInfo("reconc-eth0", "abcdef", "reconcilePod1", "PodNS1"),
+		"10.0.0.7": cns.NewPodInfo("reconc-eth0", "abcxyz", "reconcilePod2", "PodNS1"),
```

Comment:
Two ips are allocated to same interfaceID, unsupported scenario

Reply:
Can you change the interfaceId for the second ip

Reply:
Is this test passing?

Reply:
yes, it's passing